kratos/containers/geometry_container.h

```diff
@@ -1,374 +1,374 @@
+        KRATOS_ERROR_IF(i == mGeometries.end())
+            << " geometry index not found: " << GeometryName << ".";
+        return (i.base()->second);
+    }
+
+    /// Returns a reference geometry corresponding to the id
+    TGeometryType& GetGeometry(IndexType GeometryId)
+    {
+        auto i = mGeometries.find(GeometryId);
+        KRATOS_ERROR_IF(i == mGeometries.end()) << " geometry index not found: " << GeometryId << ".";
+        return *i;
+    }
+
+    /// Returns a const reference geometry corresponding to the name
+    const TGeometryType& GetGeometry(std::string GeometryName) const
+    {
+        auto hash_index = TGeometryType::GenerateId(GeometryName);
+        auto i = mGeometries.find(hash_index);
+        KRATOS_ERROR_IF(i == mGeometries.end())
+            << " geometry index not found: " << GeometryName << ".";
+        return *i;
+    }
+
+    ///@}
+    ///@name Remove Functions
+    ///@{
+
+    /// Remove the geometry with given Id from geometry container
+    void RemoveGeometry(IndexType GeometryId)
+    {
+        mGeometries.erase(GeometryId);
+    }
+
+    /// Remove the geometry with given name from geometry container
+    void RemoveGeometry(std::string GeometryName)
+    {
+        auto index = TGeometryType::GenerateId(GeometryName);
+
+        mGeometries.erase(index);
+    }
+
+    /// Remove given geometry from geometry container
+    void RemoveGeometry(TGeometryType& ThisGeometry)
+    {
+        mGeometries.erase(ThisGeometry.Id());
+    }
+
+    /// Remove given geometry from geometry container
+    void RemoveGeometry(typename TGeometryType::Pointer pThisGeometry)
+    {
+        mGeometries.erase(pThisGeometry->Id());
+    }
+
+    ///@}
+    ///@name Search Functions
+    ///@{
+
+    bool HasGeometry(IndexType GeometryId) const
+    {
+        return (mGeometries.find(GeometryId) != mGeometries.end());
+    }
+
+    bool HasGeometry(std::string GeometryName) const
+    {
+        auto hash_index = TGeometryType::GenerateId(GeometryName);
+
+        return (mGeometries.find(hash_index) != mGeometries.end());
+    }
+
+    ///@}
+    ///@name Iterators
+    ///@{
+
+    GeometryIterator GeometriesBegin()
+    {
+        return mGeometries.begin();
+    }
+
+    GeometryConstantIterator GeometriesBegin() const
+    {
+        return mGeometries.begin();
+    }
+
+    GeometryIterator GeometriesEnd()
+    {
+        return mGeometries.end();
+    }
+
+    GeometryConstantIterator GeometriesEnd() const
+    {
+        return mGeometries.end();
+    }
+
+    ///@}
+    ///@name Container Functions
+    ///@{
+
+    GeometriesContainerType& Geometries()
+    {
+        return *mGeometries;
+    }
+
+    const GeometriesContainerType& Geometries() const
+    {
+        return *mGeometries;
+    }
+
+    typename GeometriesContainerType::Pointer pGeometries()
+    {
+        return mGeometries;
+    }
+
+    void SetGeometries(typename GeometriesContainerType::Pointer pOtherGeometries)

```

Comment:
This should be removed and won't even compiled

Reply:
I removed it